### PR TITLE
Filtering based on RG output variable in PS and WFC tasks

### PR DESCRIPTION
**Pull request by Ajay-MS**

### CODE REVIEW

File: Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1 [hunk 1]
```diff
@@ -138,16 +138,17 @@ function Get-ResourceWinRmConfig
     $winrmPortToUse = ''
     $protocolToUse = ''
 
-    Write-Verbose "Starting Get-Environment cmdlet call on environment name: $environmentName" -Verbose
-    $environment = Get-Environment -environmentName $environmentName -TaskContext $distributedTaskContext
-    Write-Verbose "Completed Get-Environment cmdlet call on environment name: $environmentName" -Verbose
+    #Write-Verbose "Starting Get-Environment cmdlet call on environment name: $environmentName" -Verbose
```
Comment (kmkumaran) [inline Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1]: Can you remove the lines instead of commenting.

**Comment by pryenugu** [comment]
Could you bump up the patch version  and minimum agent version in both the tasks? Also check my comment in the agent code on how we intend to handle compatibility issues.

File: Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1 [hunk 2]
```diff
@@ -282,7 +279,7 @@ function Get-ResourceConnectionDetails
     $resourceId = $resource.Id
 
     Write-Verbose "Starting Get-EnvironmentProperty cmdlet call on environment name: $environmentName with resource id: $resourceId(Name : $resourceName) and key: $resourceFQDNKeyName" -Verbose
-    $fqdn = Get-EnvironmentProperty -EnvironmentName $environmentName -Key $resourceFQDNKeyName -TaskContext $distributedTaskContext -ResourceId $resourceId
+    $fqdn = Get-EnvironmentProperty -Environment $environment -Key $resourceFQDNKeyName -ResourceId $resourceId
```
Comment (KrishnaAdityaB) [inline Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1]: We have to update our pester tests that we maintain in VSTS Git Repo

**Comment by KrishnaAdityaB** [comment]
We have to bump up the patch version for sure.
And let's bump up the agent version later - different versions are needed based for different release branches. I will discuss with you offline

**Comment by KrishnaAdityaB** [comment]
:-1:

**Comment by kmkumaran** [comment]
Looks fine.

**Comment by KrishnaAdityaB** [comment]
Looks fine

**Pull request merged**